packages/block-library/src/video/edit.js

```diff
@@ -112,7 +122,7 @@ function VideoEdit( {
 				onReplace( embedBlock );
 				return;
 			}
-			setAttributes( { src: newSrc, id: undefined } );
+			setAttributes( { src: newSrc, id: undefined, poster: undefined } );
```

Comment:
This is a new behavior, but I think it makes the most sense for the poster to get reset / removed if a user has selected a new URL / video in the general sense.

This way, also, the poster image doesn't hide the preview of the new URL that was selected, which keeps the user from having to "play" the video to see if it worked.